crates/lib/src/qvm/mod.rs

```diff
@@ -11,6 +11,10 @@ use crate::RegisterData;
 
 mod execution;
 
+/// A map of register names (ie. "ro") to a [`RegisterData`] containing their values.
+#[allow(clippy::module_name_repetitions)]
+pub type QvmMemory = HashMap<String, RegisterData>;
```

Comment:
@randall-fulton could speak to this, but we ran into a problem a while back where the execution service was just returning data like this - a string-keyed map - and there was no room left to be able to add another top-level field.

Basically, I think that this should:

1. Have a name that accommodates data other than memory in the future (as much as it seems stable and settled now)
2. Be a struct that allows for additional fields, and so contains the `HashMap` as only one field within.

Reply:
Sure, I guess we shouldn't take the response shape for granted. I replaced it with a `QvmResultData`